liblepton/scheme/lepton/library.scm

```diff
@@ -86,29 +88,38 @@
 
 (define (source-library path)
   "Prepends the contents of given path to the default source
-library. This procedure is legacy and should be avoided in new
-code. Use set-library-contents! instead."
-  (if (string? path)
-      ;; take care of any shell variables
-      (let ((expanded-path (expand-env-variables path)))
-        (if (file-readable? expanded-path)
-            (source-library-prepend! %default-source-library
-                                     (if (absolute-file-name? expanded-path)
-                                         expanded-path
-                                         (string-append (getcwd)
-                                                        file-name-separator-string
-                                                        expanded-path)))
-            (and
-             (log! 'critical (G_ "Invalid path ~S or source not readable.\n") path)
-             #f)))
-      (and (log! 'critical (G_ "Source library path must be a string.\n")) #f)))
+library. Returns %default-source-library.
+
+This procedure is legacy and should be avoided in new code. Use
+set-library-contents! instead."
```

Comment:
How does `set-library-contents!()` replace `source-library()`?

Reply:
Currently, one default source library is used ubiquitously.  When I wrote that I thought it would be better to not depend on it but give the user a chance to make several different libs and work with them programmatically.